documentation/source/newsfragments/1380.change.rst

```diff
@@ -0,0 +1 @@
+Tests are now evaluated in order of their appearance in the :envvar:`MODULE` environment variable, their stage, and the order of invocation of the :class:`cocotb.test` decorator within a module.
```

Comment:
I think we should mention what used to happen too

Reply:
Good idea! This PR is merged, but feel free to open a follow-up PR or issue.